app/src/androidTest/java/ch/epfl/sdp/blindly/UserMapTest.kt

```diff
@@ -4,49 +4,137 @@ import android.content.Intent
-        onView(withId(R.id.map)).check(matches(withEffectiveVisibility(ViewMatchers.Visibility.VISIBLE)));
+        onView(withId(R.id.map)).check(matches(withEffectiveVisibility(ViewMatchers.Visibility.VISIBLE)))
     }
 
     @Test
-    fun openAndDisplayUserWithPoints() {
-        val intent = Intent(ApplicationProvider.getApplicationContext(), UserMapActivity::class.java)
-        intent.putExtra(POINTS, LatLng(0.0, 0.0))
-
+    fun openAndDisplayUser() {
+        val intent = Intent(
+            ApplicationProvider.getApplicationContext(),
+            UserMapActivity::class.java
+        )
+        intent.putExtra(MATCH_ID, OTHER_USER_ID)
         ActivityScenario.launch<UserMapActivity>(intent)
-        onView(withId(R.id.map)).check(matches(withEffectiveVisibility(ViewMatchers.Visibility.VISIBLE)));
+        onView(withId(R.id.map)).check(matches(withEffectiveVisibility(ViewMatchers.Visibility.VISIBLE)))
     }
+    @Test
+    fun openAndDisplayUserAndName() {
+        val intent = Intent(
+            ApplicationProvider.getApplicationContext(),
+            UserMapActivity::class.java
+        )
+        intent.putExtra(MATCH_ID, OTHER_USER_ID)
+        intent.putExtra(MATCH_NAME, OTHER_USER_NAME)
+        ActivityScenario.launch<UserMapActivity>(intent)
+        onView(withId(R.id.map)).check(matches(withEffectiveVisibility(ViewMatchers.Visibility.VISIBLE)))
+    }
+    @Test
+    fun activitySendsLocation() {
+        // Constant to check future constant completion
+        val completed = "Location received"
+        val intent = Intent(
+            ApplicationProvider.getApplicationContext(),
+            UserMapActivity::class.java
+        )
+        intent.putExtra(MATCH_ID, OTHER_USER_ID)
+        intent.putExtra(MATCH_NAME, OTHER_USER_NAME)
+        ActivityScenario.launch<UserMapActivity>(intent)
 
+        MatcherAssert.assertThat("Can't get user id", user.getUserId() != null)
+        // We wait for the callbacks to return as the test database resends the same message
+        // but from the other user
+        val future: CompletableFuture<String> = CompletableFuture()
+        database.getLocationLiveDatabase(user.getUserId()!!, OTHER_USER_ID).addListener(object :
+            DatatbaseHelper.BlindlyLiveDatabase.EventListener<BlindlyLatLng>() {
+            override fun onMessageReceived(message: Message<BlindlyLatLng>) {
+                future.complete(completed)
+            }
+            override fun onMessageUpdated(message: Message<BlindlyLatLng>) {
+                future.complete(completed)
+            }
+        })
+        assertEquals(completed, future.get())
+    }
     @Test
-    fun openAndDisplayUserWithoutPoints() {
-        val intent = Intent(ApplicationProvider.getApplicationContext(), UserMapActivity::class.java)
+    fun receiveInvalidLocation() {
+        // Constant to check future constant completion
+        val completed = "Location received"
```

Comment:
Maybe put this const in the class rather than in each test

Reply:
I agree !